src/preload/preload.js

```diff
@@ -1,141 +1,387 @@
+   * @param {Node} node 
+   * @param {string} responseText 
+   */
+  function processResponse(node, responseText) {
+    node.preloadState = node.preloadAlways ? 'READY' : 'DONE'
 
-    // Search for all child nodes that have a "preload" attribute
-    const parent = event.target || event.detail.elt;
-    parent.querySelectorAll("[preload]").forEach(function(node) {
-      // Initialize the node with the "preload" attribute
-      init(node)
+    if (getClosestAttribute(node, 'preload-images') === 'true') {
+      // Load linked resources
+      document.createElement('div').innerHTML = responseText
+    }
+  }
+
+  /**
+   * Gets attribute value from node or one of its parents
+   * @param {Node} node 
+   * @param {string} attribute 
+   * @returns { string | undefined }
+   */
+  function getClosestAttribute(node, attribute) {
+    if (node == undefined) { return undefined }
+    return node.getAttribute(attribute)
+      || node.getAttribute('data-' + attribute)
+      || getClosestAttribute(node.parentElement, attribute)
+  }
 
-      // Initialize all child elements that are anchors or have `hx-get` (use with care)
-      node.querySelectorAll('a,[hx-get],[data-hx-get]').forEach(init)
-    })
+  /**
+   * Determines if node is valid for preloading and should be
+   * initialized by setting up event listeners and handlers
+   * @param {Node} node 
+   * @returns {boolean}
+   */
+  function isValidNodeForPreloading(node) {
+    // Add listeners only to nodes which include "GET" transactions
+    // or preloadable "GET" form elements
+    const getReqAttrs = ['href', 'hx-get', 'data-hx-get'];
+    const includesGetRequest = node => getReqAttrs.some(a => node.hasAttribute(a))
+                                        || node.method === 'get';
+    const isPreloadableGetFormElement = node.form instanceof HTMLFormElement
+                                        && includesGetRequest(node.form)
+                                        && isPreloadableFormElement(node)
+    if (!includesGetRequest(node) && !isPreloadableGetFormElement) {
+      return false
+    }
+
+    // Don't preload <input> elements contained in <label>
+    // to prevent sending two requests
+    if (node.tagName instanceof HTMLInputElement && node.closest('label')) {
+      return false
+    }
+
+    return true
+  }
+
+  /**
+   * Determine if node is a form element which can be preloaded,
+   * i.e., `radio`, `checkbox`, `select` or `submit` button
+   * or a `label` of a form element which can be preloaded.
+   * @param {Node} node 
+   * @returns {boolean}
+   */
+  function isPreloadableFormElement(node) {
+    if (node instanceof HTMLInputElement) {
+      const type = node.getAttribute('type');
+      return ['checkbox', 'radio', 'submit'].includes(type);
+    }
+    if (node instanceof HTMLLabelElement) {
+      return node.control && isPreloadableFormElement(node.control);
+    }
+    return node instanceof HTMLSelectElement;
```

Comment:
I likely poorly formulated what I meant ; since we have a `instanceof HTMLInputElement` check above, and the previous code was checking `node.type === 'submit'` here, I meant that we could add a button `instanceof` check to make the code clearer on first read (so just for clarity, doesn't change the behavior).

The issue now that it has been removed, is that a `button type="submit"` would, unless I'm mistaken, not be handled anymore, as it wouldn't fall through the `instanceof HTMLInputElement` check

I guess you could either replace `if (node instanceof HTMLInputElement) {` by `if (node instanceof HTMLInputElement || node instanceof HTMLButtonElement ) {`,
or, replace
`return node instanceof HTMLSelectElement;` by `return node instanceof HTMLSelectElement || (node instanceof HTMLButtonElement && node.type === 'submit');`

If I'm not wrong and `submit` buttons were indeed not handled anymore due to the latest removal, this _might_ - unsure as I haven't reviewed the tests yet - reveal that a test case is missing to check this specific case, as it currently doesn't fail any test in the test suite,